### PR TITLE
upgrade gateway from 2.4 to 2.5

### PR DESCRIPTION
**Pull request by ccfishk**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
  part of Fix https://github.com/Kong/kubernetes-ingress-controller/issues/1684
**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR
- [ ] ensure no regression within integration tests

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -33,6 +33,11 @@
  - [0.0.5](#005---20180602)
  - [0.0.4 and prior](#004-and-prior)
 
+#### Added
```
Comment (rainest) [inline CHANGELOG.md:36]:
```suggestion
 ## [2.0.0] - TBA

#### Added
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Table of Contents
-
+-  [2.0.0-ga](#200-ga---tbd)
```
Comment (rainest) [inline CHANGELOG.md:2]:
```suggestion
-  [2.0.0](#200---tbd)
```

**Review by rainest — CHANGES_REQUESTED** [review]
Some minor fixes for the changelog (shouldn't have a suffix and needs a section header). Manifests look fine.

**Review by rainest — DISMISSED** [review]

File: config/variants/enterprise-postgres/kustomization.yaml [hunk 3]
```diff
@@ -6,4 +6,3 @@ patchesStrategicMerge:
 images:
 - name: kong
   newName: kong/kong-gateway
-  newTag: 2.4.1.0-alpine
```
Comment (rainest) [inline config/variants/enterprise-postgres/kustomization.yaml:9]: Copying from chat discussion earlier, these should continue to have their own tag declarations, even when the version here does match the version in `config/base/kong-ingress-dbless.yaml`, e.g.

```
images:
- name: kong
  newName: kong/kong-gateway
  newTag: "2.5"
```
The reason we have the image changes in these Enterprise kustomizations is that Enterprise manifests use the `kong/kong-gateway` image, not the `kong` image [used in the base Deployment](https://github.com/Kong/kubernetes-ingress-controller/blob/218be9b4b9847d626055ac2e47f02921b0ff9d56/config/base/kong-ingress-dbless.yaml#L25).

Although the `kong/kong-gateway` versions are often the same as the `kong` versions, `kong/kong-gateway` is a separate image with its own release cycle. We can and do release new `kong` tags before an equivalent `kong/kong-gateway` tag is made available, so we always want the Enterprise kustomizations to include the specific image version we want.

We don't want to have to re-add the `newTag` override in the event that we need the `kong` and `kong/kong-gateway` to differ and then take it away again if they happen to be the same.

**Review by rainest — CHANGES_REQUESTED** [review]

File: config/variants/enterprise-postgres/kustomization.yaml (hunk 3 above)
Comment (ccfishk) [reply config/variants/enterprise-postgres/kustomization.yaml:9]: Add newTag back 221ac9580a43ce3f47413ea51688a90bd6799440

File: CHANGELOG.md [hunk 4]
```diff
@@ -33,6 +33,13 @@
  - [0.0.5](#005---20180602)
  - [0.0.4 and prior](#004-and-prior)
 
+ ## [2.0.0] - TBA
+
+#### Added
+
+- upgrade kong gateway from 2.4 to 2.5
```
Comment (mflendrich) [inline CHANGELOG.md:40]:
```suggestion
- Upgraded Kong Gateway from 2.4 to 2.5
```

small editorial fix to match the style of the rest of this document

**Review by mflendrich — DISMISSED** [review]

**Pull request merged**